src/components/tx-flow/index.tsx

```diff
@@ -1,11 +1,41 @@
 import { createContext, type ReactElement, type ReactNode, useState, useEffect, useCallback } from 'react'
-import TxModalDialog from '@/components/common/TxModalDialog'
 import { useRouter } from 'next/router'
+import TxModalDialog from '@/components/common/TxModalDialog'
+import { useAppDispatch, useAppSelector } from '@/store'
+import { selectNewTxById, setNewTx, clearNewTx } from '@/store/newTxsSlice'
+
+const Flows = {
+  NewTxMenu: require('@/components/tx-flow/flows/NewTx').default,
+  RemoveModuleFlow: require('@/components/tx-flow/flows/RemoveModule').default,
+  RemoveGuardFlow: require('@/components/tx-flow/flows/RemoveGuard').default,
+  ReplaceOwnerFlow: require('@/components/tx-flow/flows/ReplaceOwner').default,
+  RemoveOwnerFlow: require('@/components/tx-flow/flows/RemoveOwner').default,
+  AddOwnerFlow: require('@/components/tx-flow/flows/AddOwner').default,
+  UpdateSafeFlow: require('@/components/tx-flow/flows/UpdateSafe').default,
+  NewSpendingLimitFlow: require('@/components/tx-flow/flows/NewSpendingLimit').default,
+  RemoveSpendingLimitFlow: require('@/components/tx-flow/flows/RemoveSpendingLimit').default,
+  ChangeThresholdFlow: require('@/components/tx-flow/flows/ChangeThreshold').default,
+  TokenTransferFlow: require('@/components/tx-flow/flows/TokenTransfer').default,
+  RejectTx: require('@/components/tx-flow/flows/RejectTx').default,
+  SuccessScreen: require('@/components/tx-flow/flows/SuccessScreen').default,
+  SafeAppsTxFlow: require('@/components/tx-flow/flows/SafeAppsTx').default,
+  SignMessageFlow: require('@/components/tx-flow/flows/SignMessage').default,
+  SignMessageOnChainFlow: require('@/components/tx-flow/flows/SignMessageOnChain').default,
+  ExecuteBatchFlow: require('@/components/tx-flow/flows/ExecuteBatch').default,
+  ConfirmTxFlow: require('@/components/tx-flow/flows/ConfirmTx').default,
+  ReplaceTxMenu: require('@/components/tx-flow/flows/ReplaceTx').default,
+  NftTransferFlow: require('@/components/tx-flow/flows/NftTransfer').default,
+}
 
 const noop = () => {}
 
+export type TxFlow = {
+  component: keyof typeof Flows
+  props?: any
```

Comment:
@iamacook can you remind me how you guys made this type safe originally? Before we pivoted to defining the current flow as a JSX element.

Reply:
You'll need to cast every import in `Flows` to their respective component as `require` imports are `any`, then you can extract their props with the helper type of React, `ComponentProps`, something like:

```ts
type TxFlow<T extends keyof typeof Flows> = {
  component: T
  props: ComponentProps<typeof Flows[T]>
}
```